### ISSUE TITLE
introduce refreshToken to keep user logged in from browser 

### ISSUE THREAD

**Issue by ayan-biswas0412** [CircuitVerse/CircuitVerse#2112]
**Is your feature request related to a problem? Please describe.**
In most of the websites we see after log in for the first time it keeps user in logged in state until and unless any unusual activity happens from user's account but in the CircuitVerse website it not happens , one user has to give his/her credential every time he/she visits the website.

**Describe the solution you'd like**
I would like to update the backend side the login api , when one user will login/signup for the first time it will return one ```refreshToken``` along with the ```token``` now we have to store the refreshToken in local storage and everytime user visit the website refreshToken will be exchanged to get a new token + refreshToken and this process continues .
If any unusual activity happens refreshToken will be automatically revoked .

**Describe alternatives you've considered**
Alternative solution is already running in CircuitVerse which is the current login/signup flow which is very fine , I just want to enhance the user experience 

**Additional context**
The proper implementation can be done using the google auth API and here is the reference guide [here](https://firebase.google.com/docs/reference/rest/auth)

the previous login/signup api will remain as it is only change is needed to return the refreshToken in login/signup both cases 
and also we have to write another api for exchanging between refreshToken and token

<img width="918" alt="image" src="https://user-images.githubusercontent.com/52851184/107786734-3eab9c00-6d74-11eb-8ea4-b9853e24066d.png">

**Are you working on this? (Yes/No)**

No, but looking for approval from mentors and other contributors , if I get approval then I can implement this

**Comment by ayan-biswas0412** [comment]
would like to hear suggestions from you guys
@satu0king  @sal2701 @Shivansh2407 @nitin10s

**Comment by Dnouv** [comment]
Hello @ayan-biswas0412 
Is this issue still open for contributions? If it is, please let me know. I would like to contribute to this issue.
Thank you!

**Comment by ayan-biswas0412** [comment]
> Hello @ayan-biswas0412 Is this issue still open for contributions? If it is, please let me know. I would like to contribute to this issue. Thank you!

Yeah go ahead, but please research on it a bit, because this issue needs a proper pathway to proceed, so before coding you can research a bit and write an probable implementation pathway here

**Comment by Dnouv** [comment]
Thank you for the prompt reply; I'll look up more details about the implementation strategy and will post here. The starting point would most probably be looking for the part number for the components; I've done introductory minor courses in Electronics; however, it might require some brushup.
I hope to receive your guidance if got stuck somewhere.
Thank you!

**Comment by Dnouv** [comment]
@ayan-biswas0412  I've planned to introduce a new node in the Object `elementHierarchy` of the elements in addition to the `name` and `label` node say `type` or `number`, then make a list of these IC numbers available to the simulator as done for the elements name. Proceeding forward in the search function of elements in addition to filtering the array-based only on `name` the new function will also filter based on the components' IC number. 
As for the function, I'm planning to use the JS array `some` method. So, this will run the search for components on basis of the number in parallel to the name.
I do have a few queries while looking up the IC number I found there were two types CMOS and TTL and their respective IC codes are unique. So which one would we prefer? Please share your thoughts.

Thank you!

**Comment by Dnouv** [comment]
Hello @ayan-biswas0412 
I have raised a PR with the required changes, could you please review it, and provide feedback. If any changes are required, please let me know I will make the changes accordingly.
Thank you!
